File: LeetCode/Easy/0389-find-the-difference/0389-find-the-difference.java
class Solution {
    public char findTheDifference(String s, String t) {

        if(t.length() <= 0) return '0';
        if(t.length() <= 1) return t.charAt(0);

        char[] tChar = t.toCharArray();
        Arrays.sort(tChar);

        int[] dictionary = new int[26];

        for(char c : s.toCharArray()) {
            dictionary[c - 'a']++;
        }

        int accumulate = 0;

        System.out.println(new String(tChar));

        for(int i = 0; i < dictionary.length; ++i) {
            accumulate += dictionary[i];
            if (tChar[accumulate - 1] - 'a' != i || tChar[accumulate] - 'a' == i)
                return tChar[accumulate];
        }

        return '0';
    }
}

File: LeetCode/Easy/0389-find-the-difference/0389-find-the-difference.kt
class Solution {
    fun findTheDifference(s: String, t: String): Char {
        if(t.length <= 0) return '0';
        if(t.length <= 1) return t[0];
        // println("#####################");
        var dictionary = IntArray(26, {0});
        var tChar = t.toCharArray();
        tChar.sort();

        for (c in s.toCharArray()) {
            dictionary[c - 'a']++;
            // println(dictionary[c - 'a']);
        }

        var accum = 0;
        // println("#####################");
        dictionary.forEachIndexed { index, count -> 
            // print(accum);
            // print(" / ");
            // println(index);
            accum += count;
            if (tChar[accum - 1] - 'a' != index || tChar[accum] - 'a' == index)
                return tChar[accum];
        }

        return '0';
    }
}